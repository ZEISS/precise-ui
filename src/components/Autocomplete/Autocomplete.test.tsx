import * as React from 'react';
import * as enzyme from 'enzyme';
import { Autocomplete } from './';

describe('<Autocomplete />', () => {
  it('should render <Autocomplete> component', () => {
    const wrapper = enzyme.shallow(<Autocomplete suggestions={['one', 'two', 'three']} defaultValue="five" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show suggestions list when focused', () => {
    const wrapper = enzyme.mount(<Autocomplete suggestions={['one', 'two', 'three']} defaultValue="five" />);
    // tslint:disable-next-line:no-string-literal
    global['cancelAnimationFrame'] = jest.fn();
    wrapper
      .find('div')
      .at(0)
      .prop('onFocus')({} as React.FocusEvent);
    expect(wrapper.update().find('li')).toHaveLength(3);

    // tslint:disable-next-line:no-string-literal
    delete global['cancelAnimationFrame'];
  });
});
