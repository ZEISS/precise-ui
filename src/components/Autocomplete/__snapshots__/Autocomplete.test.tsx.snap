// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Autocomplete /> should render <Autocomplete> component 1`] = `
<AutocompleteInt
  defaultValue="five"
  suggestions={
    Array [
      "one",
      "two",
      "three",
    ]
  }
/>
`;
